File: app/src/main/java/com/qadam/lastpuff/util/TimeFormatUtils.kt
package com.qadam.lastpuff.util

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.concurrent.TimeUnit

object TimeFormatUtils {
    private val timeFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
    private val dateTimeFormat = SimpleDateFormat("dd.MM HH:mm", Locale.getDefault())

    fun formatTime(timestamp: Long): String = timeFormat.format(Date(timestamp))

    fun formatDateTime(timestamp: Long): String = dateTimeFormat.format(Date(timestamp))

    fun formatAgo(timestamp: Long, now: Long = System.currentTimeMillis()): String {
        val diff = now - timestamp
        if (diff < 0) return "только что"
        val minutes = TimeUnit.MILLISECONDS.toMinutes(diff)
        val hours = TimeUnit.MILLISECONDS.toHours(diff)
        val days = TimeUnit.MILLISECONDS.toDays(diff)
        return when {
            minutes < 1 -> "только что"
            minutes < 60 -> "$minutes мин. назад"
            hours < 24 -> "$hours ч. назад"
            else -> "$days дн. назад"
        }
    }

    fun intensityStars(intensity: Int): String =
        "★".repeat(intensity.coerceIn(1, 5)) + "☆".repeat((5 - intensity).coerceAtLeast(0))
}

File: ios/QadamLastPuff/Utils/TimeFormatUtils.swift
import Foundation

enum TimeFormatUtils {
    private static let timeFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "HH:mm"
        return f
    }()

    private static let dateTimeFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "dd.MM.yyyy HH:mm"
        return f
    }()

    static func formatTime(_ date: Date) -> String {
        timeFormatter.string(from: date)
    }

    static func formatDateTime(_ date: Date) -> String {
        dateTimeFormatter.string(from: date)
    }

    static func formatAgo(_ date: Date, now: Date = Date()) -> String {
        let diff = now.timeIntervalSince(date)
        guard diff >= 0 else { return "только что" }
        let minutes = Int(diff / 60)
        let hours = Int(diff / 3600)
        let days = Int(diff / 86400)
        if minutes < 1 { return "только что" }
        if minutes < 60 { return "\(minutes) мин. назад" }
        if hours < 24 { return "\(hours) ч. назад" }
        return "\(days) дн. назад"
    }

    static func intensityStars(_ intensity: Int) -> String {
        let clamped = min(max(intensity, 1), 5)
        return String(repeating: "★", count: clamped) + String(repeating: "☆", count: 5 - clamped)
    }
}
